File: NodeWorker/src/models/node_execution_request.py
from src.models.message import Message


class NodeExecutionRequest(Message):

    def __init__(self, deserialized: dict):
        super().__init__()
        self.pipeline_id: str = deserialized['PipelineId']
        self.execution_id: str = deserialized['ExecutionId']
        self.node_id = deserialized['NodeId']
        self.operation_id: str = deserialized['OperationId']
        self.operation_name: str = deserialized['OperationName']
        self.result_key: str = deserialized['ResultKey']

    def get_pipeline_id(self) -> str:
        return self.pipeline_id

    def get_node_id(self) -> str:
        return self.node_id

    def get_execution_id(self) -> str:
        return self.execution_id

    def get_operation_id(self):
        return self.operation_id

    def get_operation_name(self):
        return self.operation_name

    def get_result_key(self):
        return self.result_key

File: PipelineService/Models/MqttMessages/NodeExecutionRequest.cs
using System;

namespace PipelineService.Models.MqttMessages
{
	public abstract class NodeExecutionRequest : BaseMqttMessage
	{
		/// <summary>
		/// The pipeline's id the node was executed for.
		/// </summary>
		public Guid PipelineId { get; set; }

		/// <summary>
		/// The pipeline execution this node belongs to. 
		/// </summary>
		public Guid ExecutionId { get; set; }

		/// <summary>
		/// The node's id that was executed.
		/// </summary>
		public Guid NodeId { get; set; }

		/// <summary>
		/// The operations id that will be executed.
		/// </summary>
		public Guid OperationId { get; set; }

		/// <summary>
		/// The name of the operation to make messages more readable for humans.
		/// </summary>
		public string OperationName { get; set; }

		/// <summary>
		/// The key the resulting dataset will be stored as.
		/// </summary>
		public string ResultKey { get; set; }
	}
}
